app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/standard/impl/AppEngineStandardCommunityWebIntegration.java

```diff
@@ -49,10 +52,27 @@
   @Override
   public VirtualFile suggestParentDirectoryForAppEngineWebXml(
       @NotNull Module module, @NotNull ModifiableRootModel rootModel) {
-    final VirtualFile root = ArrayUtil.getFirstElement(rootModel.getContentRoots());
+    BuildSystemAppEngineWebXmlDirectoryProvider[] appEngineWebXmlDirectoryProviders =
+        Extensions.getExtensions(BuildSystemAppEngineWebXmlDirectoryProvider.EP_NAME);
+    for (BuildSystemAppEngineWebXmlDirectoryProvider provider : appEngineWebXmlDirectoryProviders) {
+      Optional<String> pathOptional = provider.getAppEngineWebXmlDirectoryPath(module);
+
+      if (pathOptional.isPresent()) {
+        try {
+          return VfsUtil.createDirectoryIfMissing(pathOptional.get());
+        } catch (IOException ioe) {
+          LOG.warn(
+              "Exception attempting to create appengine-web.xml in location specified by build-aware extension",
+              ioe);
+        }
+      }
+    }
+
+    // Build-aware strategies failed or missing, fall back to simple native approach
+    VirtualFile root = ArrayUtil.getFirstElement(rootModel.getContentRoots());
     if (root != null) {
       try {
-        return VfsUtil.createDirectoryIfMissing(root, "WEB-INF");
+        return VfsUtil.createDirectoryIfMissing(root, "web/WEB-INF");
```

Comment:
What was the reason for this change, from plain `WEB-INF` to adding `web/`?

Reply:
this code flow is for community edition native (no build system) projects. In this case, I am simply mimicking the directory structure that intellij generates for its web applications (when using the wizard):

![selection_787](https://user-images.githubusercontent.com/1735744/42821844-7d1a7028-89a7-11e8-92ad-9ed70c9f6f4b.png)

Note that `web/WEB-INF` is under the root.

Previously, it was omitting the web directory and just placing `WEB-INF` under root

Reply:
Ah I see what you mean by "native" in the comment then. Could we possibly expand the "native" in the comment as well and say we use default IDEA directory structure in this case? Might be helpful some time after for easier maintenance.

Reply:
Nit: possibly make it even easier with well named constant for `web`?

Reply:
sounds good (both points).

Reply:
done